### PR TITLE
Homework4: Dmitry Makarevich

### PR DESCRIPTION
**Pull request by DzmitryMak**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle).
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks.
- [ ] my PR made from the fork feature branch to appropriate branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework 4/task2.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+n = int(input('Введите количество стран '))
+
+dct = {}
+answ = ''
+
+for _ in range(n):
+    lst = (input('Страна и города ')).split()
+    dct[lst[0]] = lst[1:]
```
Comment (belovmd) [inline src/homework 4/task2.py]: a, *b = smth.split()

File: src/homework 4/task3.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"""
+Даны два списка чисел. Посчитайте, сколько различных чисел
+содержится одновременно как в первом списке, так и во втором.
+"""
+
+lst1 = [1, 3, 4, 5, 7, 8, 23, 11]
+lst2 = [3, 4, 5, 6, 8, 21, 11, 9]
+
+print(len(set(lst1) ^ set(lst2)), 'различных чисел в списках')
```
Comment (belovmd) [inline src/homework 4/task3.py]: doesn't solve task in description

File: src/homework 4/task4.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+"""
+Даны два списка чисел. Посчитайте, сколько различных
+чисел входит только в один из этих списков
+"""
+
+lst1 = [1, 3, 4, 5, 7, 8, 23, 11]
+lst2 = [3, 4, 5, 6, 8, 21, 11, 9]
+
+print(len(set(lst1) - set(lst2)), 'различных чисел в один список')
```
Comment (belovmd) [inline src/homework 4/task4.py]: doesn't solve task in description

File: src/homework 4/task7.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+a = 75
+b = 39
+
+while a and b:
+    if a > b:
+        a = a % b
+    else:
+        b = b % a
+
+print('НОД =', a + b)
```
Comment (belovmd) [inline src/homework 4/task7.py]: it's a cheat - you just don't know where is 0 and where is the result.

**Review by belovmd — CHANGES_REQUESTED** [review]

File: src/homework 4/task7.py (hunk 4 above)
Comment (DzmitryMak) [reply src/homework 4/task7.py]: fixed

File: src/homework 4/task4.py (hunk 3 above)
Comment (DzmitryMak) [reply src/homework 4/task4.py]: fixed

File: src/homework 4/task3.py (hunk 2 above)
Comment (DzmitryMak) [reply src/homework 4/task3.py]: fixed

File: src/homework 4/task2.py (hunk 1 above)
Comment (DzmitryMak) [reply src/homework 4/task2.py]: fixed

**Review by belovmd — APPROVED** [review]
9

**Pull request closed without merging**